engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1657,6 +1657,10 @@ protected boolean sendStop(final VirtualMachineGuru guru, final VirtualMachinePr
                     return false;
                 }
 
+                final UserVmVO userVm = _userVmDao.findById(vm.getId());
+                userVm.setPowerState(PowerState.PowerOff);
+                _userVmDao.update(userVm.getId(), userVm);
+
                 guru.finalizeStop(profile, answer);
```

Comment:
@Spaceman1984 should we update the power state after guru finalises the stop? What is there is an exception then the power state will be off while VM may actually be running?

Reply:
np @rhtyd, I'll move it down.